client/my-sites/themes/auto-loading-homepage-modal.jsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import {
+	getCanonicalTheme,
+	hasActivatedTheme,
+	themeHasAutoLoadingHomepage,
+	isActivatingTheme,
+	isThemeActive,
+	shouldShowHomepageWarning,
+	getPreInstallingThemeId,
+} from 'state/themes/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	acceptAutoLoadingHomepageWarning,
+	hideAutoLoadingHomepageWarning,
+	activate as activateTheme,
+} from 'state/themes/actions';
+
+/**
+ * Style dependencies
+ */
+import './auto-loading-homepage-modal.scss';
+
+class AutoLoadingHomepageModal extends Component {
+	static propTypes = {
+		source: PropTypes.oneOf( [ 'details', 'list', 'upload' ] ).isRequired,
+		theme: PropTypes.shape( {
+			author: PropTypes.string,
+			author_uri: PropTypes.string,
+			id: PropTypes.string,
+			name: PropTypes.string,
+		} ),
+		hasActivated: PropTypes.bool.isRequired,
+		isActivating: PropTypes.bool.isRequired,
+		hasAutoLoadingHomepage: PropTypes.bool,
+		siteId: PropTypes.number,
+		isVisible: PropTypes.bool,
+		onClose: PropTypes.func,
+		installingThemeId: PropTypes.string.isRequired,
+	};
+
+	onCloseModal = ( activate = false ) => () => {
+		if ( activate ) {
+			const { installingThemeId, siteId, source } = this.props;
+			this.props.acceptAutoLoadingHomepageWarning( installingThemeId );
+			return this.props.activateTheme( installingThemeId, siteId, source );
+		}
+		this.props.hideAutoLoadingHomepageWarning();
+	};
+
+	render() {
+		const {
+			theme,
+			hasActivated,
+			isActivating,
+			hasAutoLoadingHomepage,
+			isCurrentTheme,
+			isVisible = false,
+		} = this.props;
+
+		// Nothing to do when it's the current theme.
+		if ( isCurrentTheme ) {
+			return null;
+		}
+
+		// Nothing to show if the theme doesn't have auto loading homepage.
+		if ( ! hasAutoLoadingHomepage ) {
+			return null;
+		}
+
+		// Hide while is activating or when it's activated.
+		if ( isActivating || hasActivated ) {
+			return null;
+		}
+
+		if ( ! theme ) {
+			return null;
+		}
+
+		const { name: themeName } = this.props.theme;
+
+		return (
+			<Dialog
+				className="themes__auto-loading-homepage-modal"
+				isVisible={ isVisible }
+				buttons={ [
+					{
+						action: 'keepCurrentTheme',
+						label: translate( 'No, keep my current theme' ),
+						isPrimary: false,
+						onClick: this.onCloseModal( false ),
+					},
+					{
+						action: 'activeTheme',
+						label: translate( 'Yes, active %(themeName)s', {
+							args: { themeName },
+						} ),
+						isPrimary: true,
+						onClick: this.onCloseModal( true ),
+					},
+				] }
+				onClose={ this.onCloseModal( false ) }
```

Comment:
I might be missing something but those `onCloseModal` are not passing function references but are rather being invoked immediately. Is that intentional or is it possibly missing `() =>` prefix?

Reply:
The `this.onCloseModal()` method is already [returning the function handle](https://github.com/Automattic/wp-calypso/pull/39436/files#diff-dcbe1254be3df4a64783d3aa85867606R52)r.

Reply:
Anyway, [I've renamed the method](https://github.com/Automattic/wp-calypso/pull/39436/commits/b0e169a672ab5607b2f01b717eb81c08522470de) since I like the idea to connect the event (which is the action, a verb, at the left side of the line) with the handler (a noun, right side):

```es6
<Dialog
  onClose={ closeModalHandler }
/>
```